### PR TITLE
Add In-App Messaging Getting Started (Developer Preview)

### PR DESCRIPTION
**Pull request by calebpollman**
_Issue #, if available:_
NA

_Description of changes:_
Add the Developer Preview Getting Started guide for In-App Messaging for JS and React Native

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by offlineprogrammer — APPROVED** [review]

**Review by Ashish-Nanda — APPROVED** [review]

File: src/fragments/lib/in-app-messaging/js/create-campaigns.mdx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+   - **Secondary button**: Allows the addition of an extra button for additional functionality.
+   - **Custom Data**: Allows the in-app message to pass additional data to the frontend app once it is triggered by an event.
+
+   <Callout informational>
+      As React Native does not support SVG rendering out of the box, Amplify cannot render SVG images by default. For SVG image support with In-App Messaging a custom UI implementation is required.
+   </Callout>
+
+   For this tutorial you can create a simple message as shown below. Customers in your application will see the same message once the event is triggered.
+
+   ![In-App-Messaging](/images/lib/in-app-messaging/create-in-app-message.png)
```
Comment (pananapread) [inline src/fragments/lib/in-app-messaging/js/create-campaigns.mdx:40]: broken image 
image

File: src/fragments/lib/in-app-messaging/js/getting-started.mdx [hunk 2]
```diff
@@ -0,0 +1,125 @@
+Make sure you have completed the below steps:
+
+- [Completed the pre-requisite steps.](/lib/in-app-messaging/prerequisites)
```
Comment (pananapread) [inline src/fragments/lib/in-app-messaging/js/getting-started.mdx:3]: prerequisite

File: src/fragments/lib/in-app-messaging/js/getting-started.mdx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+- [Create a campaign from the pinpoint console.](/lib/in-app-messaging/create-campaigns)
+
+## Integrate your application
+
+In your application directory, you should first install the necessary dependencies. Although Amplify In-App Messaging can be used as a standalone JavaScript library, the only UI integration currently available is only for _React Native_. The integrations steps here assume a React Native application as the implementation target.
```
Comment (pananapread) [inline src/fragments/lib/in-app-messaging/js/getting-started.mdx:8]: Although Amplify In-App Messaging can be used as a standalone JavaScript library, the ~only~ UI integration **is** currently available ~is~ only for _React Native_.

File: src/fragments/lib/in-app-messaging/js/getting-started.mdx [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+```bash
+npm install -E aws-amplify@in-app-messaging aws-amplify-react-native@in-app-messaging amazon-cognito-identity-js @react-native-community/netinfo @react-native-async-storage/async-storage @react-native-picker/picker react-native-get-random-values react-native-url-polyfill
+```
+
+You will also need to install pod dependencies for `iOS`. Navigate the the ios directory in your project and run the install command.
```
Comment (pananapread) [inline src/fragments/lib/in-app-messaging/js/getting-started.mdx:14]: Navigate **to** the **iOS** directory in your project and run the install command.

File: src/fragments/lib/in-app-messaging/js/sync-messages.mdx [hunk 5]
```diff
@@ -0,0 +1,10 @@
+The first step to getting in-app messages in front of your app’s active users is to make sure all available In-App Messaging campaign messages are synced to their device, which would be triggered when an analytics or In-App Messaging event is triggered. Amplify In-App Messaging empowers you with the control of when and how often you perform this sync.
```
Comment (pananapread) [inline src/fragments/lib/in-app-messaging/js/sync-messages.mdx:1]: Run-on sentence! Suggested:

To trigger messages, you must sync them from your In-App Messaging campaigns to your active user's devices. These messages are then triggered with an analytics or In-App Messaging event. You can control when and how often this sync is performed

File: src/fragments/lib/in-app-messaging/js/sync-messages.mdx [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+
+```js
+await InAppMessaging.syncMessages();
+```
+
+<Callout informational>
+  NOTE: Syncing messages will always overwrite existing messages currently on
+  their device so they are always up to date when the sync is performed.
```
Comment (pananapread) [inline src/fragments/lib/in-app-messaging/js/sync-messages.mdx:9]: NOTE: Syncing messages will always overwrite existing messages currently on
a user's device so they are always up to date when the sync is performed.

File: src/fragments/lib/in-app-messaging/js/display-message.mdx [hunk 7]
```diff
@@ -0,0 +1,33 @@
+In-app messages are displayed when an In-App Messaging or analytics event is sent and matches the criteria set forth by your active In-App Messaging campaigns.
+
+## Analytics event
+
+Now that messages have been synced to your users’ devices, Amplify In-App Messaging will allow you to start displaying them with Amplify Analytics events with no additional integration steps. Any events you record or are already recording using the `Analytics.record` API are automatically picked up and processed by In-App Messaging. If the event matches the attributes and criteria defined in an in-app message, that message will be displayed.
```
Comment (pananapread) [inline src/fragments/lib/in-app-messaging/js/display-message.mdx:5]: Now that messages have been synced to your users’ devices, Amplify In-App Messaging allows you to start displaying them through Amplify Analytics events with no additional integration steps.

File: src/directory/directory.js [hunk 8]
```diff
@@ -450,6 +450,66 @@ const directory = {
...
+            route: "/lib/in-app-messaging/create-campaigns",
+            filters: ["js"],
+          },
+          {
+            title: "Getting started",
+            route: "/lib/in-app-messaging/getting-started",
+            filters: ["js"],
+          },
+          {
+            title: "Sync messages",
```
Comment (calebpollman) [inline src/directory/directory.js:477]:
```suggestion
            title: "Synchronize messages",
```

File: src/fragments/lib/in-app-messaging/js/display-message.mdx (hunk 7 above)
Comment (pananapread) [inline src/fragments/lib/in-app-messaging/js/display-message.mdx:5]:
```suggestion
Now that messages have been synced to your users’ devices, Amplify In-App Messaging will allow you to start displaying them with Amplify Analytics events with no additional integration steps. Any events you record or are already recording using the `Analytics.record` API are automatically picked up and processed by In-App Messaging. If the event matches the attributes and criteria defined in an in-app message, that message will be displayed.
```

File: src/fragments/lib/in-app-messaging/js/sync-messages.mdx (hunk 6 above)
Comment (pananapread) [inline src/fragments/lib/in-app-messaging/js/sync-messages.mdx:9]:
```suggestion
a user's device so they are always up to date when the sync is performed.
```

File: src/fragments/lib/in-app-messaging/js/getting-started.mdx [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+Next, you should import and load the `aws-exports.js` configuration file created and modified in an earlier step into your application. Add the Amplify configuration step to your application’s root entry point. For example, `index.js`.
+
+```js
+import Amplify from 'aws-amplify';
+import awsconfig from './src/aws-exports';
+
+Amplify.configure(awsconfig);
+```
+
+Finally, integrate the Amplify React Native UI components for In-App Messaging in your application’s root component. For example, `App.jsx`.
```
Comment (calebpollman) [inline src/fragments/lib/in-app-messaging/js/getting-started.mdx:42]:
```suggestion
Finally, integrate the Amplify React Native UI components for In-App Messaging in your application’s root component. For example, `App.js`.
```

File: src/fragments/lib/in-app-messaging/js/sync-messages.mdx (hunk 6 above)
Comment (calebpollman) [reply src/fragments/lib/in-app-messaging/js/sync-messages.mdx:9]:
```suggestion
  the user's device so they are always up to date when the sync is performed.
```

File: src/fragments/lib/in-app-messaging/js/getting-started.mdx [hunk 10]
```diff
@@ -0,0 +1,125 @@
+Make sure you have completed the below steps:
+
+[Completed the prerequisite steps.](/lib/in-app-messaging/prerequisites)
+- [Create a campaign from the pinpoint console.](/lib/in-app-messaging/create-campaigns)
```
Comment (calebpollman) [inline src/fragments/lib/in-app-messaging/js/getting-started.mdx:4]:
```suggestion
- [Completed the prerequisite steps.](/lib/in-app-messaging/prerequisites)
- [Created a campaign from the pinpoint console.](/lib/in-app-messaging/create-campaigns)
```

**Review by cshfang — APPROVED** [review]

**Review by pananapread — APPROVED** [review]

**Review by abdallahshaban557 — APPROVED** [review]

**Pull request merged**